text/0000-naming-conventions.md

```diff
@@ -0,0 +1,267 @@
+- Start Date: 2018-03-21
+- RFC PR: (leave this empty)
+- WordPress Coding Standards Issue: (leave this empty)
+
+# Summary
+
+This guide is intended to maximize consistency when writing code in the WordPress environment.
+Naming your language constructs is important and should reflect the nature of what it does.
+
+# Basic example
+
+Naming variables, functions, classes, namespaces should be unified across the core, themes and plugins.
+
+Variables should be clear and understandable, written in lowercase using hyphens as a separator.
+
+```php
+$post_array;
+```
+
+Classes (namespaces, traits, interfaces, abstract classes and classes) should be written in capitalized words separated by underscores.
+
+```php
+namespace Theme_Name\Subnamespace;
+
+class Class_Name implements Interface {
+  // class implementation.
+}
+```
+
+PHP constants should be written in lowercase, while user defined constants should be written in uppercase separated by underscores.
+
+```php
+define( 'DOING_AJAX', false );
+```
+
+File names should be written using lowercase letters separated by hyphens, and class file names must be based on the class name, with the underscores in the class name replaced by hyphens.
+
```

Comment:
What about interface and trait file names ?

Reply:
What about namespaces, if you have something like `MyPlugin\ComponentA\Controller` and `MyPlugin\ComponentB\Controller` ?